tests/db/test_controls_workflows_mapping.py

```diff
@@ -0,0 +1,176 @@
+
+        assert control.control_id == control_id
+        assert control.control_name == "Test Control"
+        assert control.control_description == "A test control"
+        assert control.user_id == TEST_USER_ID
+
+
+@pytest.mark.asyncio
+async def test_create_workflow(store: AdmyralStore):
+    async with store.async_session_maker() as session:
+        workflow_id = generate_random_workflow_id()
+        workflow = await create_workflow(session, workflow_id)
+
+        assert workflow.workflow_id == workflow_id
+        assert workflow.workflow_name == f"Test Workflow-{workflow_id}"
+        assert workflow.user_id == TEST_USER_ID
+
+
+@pytest.mark.asyncio
+async def test_create_control_workflow_mapping(store: AdmyralStore):
+    async with store.async_session_maker() as session:
+        control, workflow, mapping = await setup_scenario(session)
+
+        # Verify the mapping
+        assert mapping.control_id == control.control_id
+        assert mapping.workflow_id == workflow.workflow_id
+
+        # Refresh the relationships
+        await session.refresh(control, ["control_workflow_mappings"])
+        await session.refresh(workflow, ["control_workflow_mappings"])
+
+        # Get the related objects through the mapping
+        control_mappings = control.control_workflow_mappings
+        workflow_mappings = workflow.control_workflow_mappings
+
+        assert len(control_mappings) == 1
+        assert len(workflow_mappings) == 1
+        assert control_mappings[0].workflow_id == workflow.workflow_id
+        assert workflow_mappings[0].control_id == control.control_id
+
+
+@pytest.mark.asyncio
+async def test_cascade_delete_workflow(store: AdmyralStore):
+    async with store.async_session_maker() as session:
+        await store.clean_up_workflow_data_of(TEST_USER_ID)
+        await store.clean_up_controls_data(TEST_USER_ID)
+        _, workflow, _ = await setup_scenario(session)
+
+        # Create a control result
+        control_result = WorkflowControlResultsSchema(
+            workflow_id=workflow.workflow_id,
+            run_id="run123",
+            result=True,
+            user_id=TEST_USER_ID,
+        )
+        session.add(control_result)
+        await session.commit()
+
+        # Verify everything is created
+        mapping_result = await session.exec(select(ControlsWorkflowsMappingSchema))
+        mappings = mapping_result.all()
+        assert len(mappings) == 1
+
+        results_result = await session.exec(select(WorkflowControlResultsSchema))
+        results = results_result.all()
+        assert len(results) == 1
+
+        # First delete the control results
+        results_to_delete = await session.exec(
+            select(WorkflowControlResultsSchema).where(
+                WorkflowControlResultsSchema.workflow_id == workflow.workflow_id
+            )
+        )
+        for result in results_to_delete.all():
+            await session.delete(result)
```

Comment:
You can just do

```suggestion
        await session.exec(
            delete(WorkflowControlResultsSchema).where(
                WorkflowControlResultsSchema.workflow_id == workflow.workflow_id
            )
        )
```

There is no need for retrieving everything first.